libdevice/wrapper.h

```diff
@@ -25,9 +25,10 @@ void *__devicelib_memset(void *dest, int c, size_t n);
 DEVICE_EXTERN_C
 int __devicelib_memcmp(const void *s1, const void *s2, size_t n);
 DEVICE_EXTERN_C
-void __devicelib_assert_fail(const char *expr, const char *file, int32_t line,
-                             const char *func, uint64_t gid0, uint64_t gid1,
-                             uint64_t gid2, uint64_t lid0, uint64_t lid1,
-                             uint64_t lid2);
+__attribute__((noinline, optnone)) void
```

Comment:
Can you add `naked` to the attributes?
There are two optimizations that cause incorrect behaviors: inlining and dead argument elimination.
Interestingly, `optnone` does not block dead argument elimination, but `naked` does.

Reply:
I added it, but there were compilation errors:  error: non-ASM statement in naked function is not supported.